packages/embed/src/utils/change-color-opacity.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { changeColorOpacity } from './change-color-opacity'
+
+describe('#changeColorOpacity', () => {
+  it('should change opacity to 255 by default', () => {
+    expect(changeColorOpacity('rgba(255, 125, 0, 0)')).toBe('rgba(255, 125, 0, 255)')
+  })
+
+  it('should change opacity to specified value', () => {
+    expect(changeColorOpacity('rgba(255, 125, 0, 50)', 25)).toBe('rgba(255, 125, 0, 25)')
+  })
+
+  it('should change decimal opacity to specified value', () => {
+    expect(changeColorOpacity('rgba(255, 125, 0, 0.5)', 125)).toBe('rgba(255, 125, 0, 125)')
+    expect(changeColorOpacity('rgba(255, 125, 0, .250)', 125)).toBe('rgba(255, 125, 0, 125)')
+  })
+
+  it('should not change opacity for color without opacity', () => {
+    expect(changeColorOpacity('rgb(255, 125, 0)', 25)).toBe('rgb(255, 125, 0)')
+    expect(changeColorOpacity('#ff0000', 25)).toBe('#ff0000')
+  })
+})
```

Comment:
wouldn't be more clear in a test.each loop?

Reply:
What do you have in mind? Something like this?

```
import { changeColorOpacity } from './change-color-opacity'

describe('#changeColorOpacity', () => {
  const scenarios = [
    {
      title: 'should change opacity to 255 by default',
      inputColor: 'rgba(255, 125, 0, 0)',
      inputOpacity: undefined,
      output: 'rgba(255, 125, 0, 255)',
    },

    {
      title: 'should change opacity to specified value',
      inputColor: 'rgba(255, 125, 0, 50)',
      inputOpacity: 25,
      output: 'rgba(255, 125, 0, 25)',
    },

    {
      title: 'should change decimal opacity to specified value',
      inputColor: 'rgba(255, 125, 0, 0.5)',
      inputOpacity: 125,
      output: 'rgba(255, 125, 0, 125)',
    },

    {
      title: 'should change opacity to specified value',
      inputColor: 'rgba(255, 125, 0, .250)',
      inputOpacity: 125,
      output: 'rgba(255, 125, 0, 125)',
    },

    {
      title: 'should not change opacity for color without opacity',
      inputColor: 'rgb(255, 125, 0)',
      inputOpacity: 25,
      output: 'rgb(255, 125, 0)',
    },

    { title: 'should change opacity to specified value', inputColor: '#ff0000', inputOpacity: 25, output: '#ff0000' },
  ]

  scenarios.forEach(({ title, inputColor, inputOpacity, output }) => {
    it(title, () => {
      expect(changeColorOpacity(inputColor, inputOpacity)).toBe(output)
    })
  })
})
```

Reply:
yep 👍  WDYT?

Reply:
I think it adds more abstraction to the test and makes it harder to read. If you write multiple `expect`s it is something every developer understands. If you build an array like this it is not obvious at first look what is going on.

I think this approach would be valid for a simpler scenarios, eg. [like this one](https://github.com/Typeform/embed/blob/dist1607/packages/embed/src/utils/is-defined.spec.ts#L15-L23).